packages/firestore/src/api/user_data_converter.ts

```diff
@@ -273,20 +287,44 @@ export class DocumentKeyReference {
 export class UserDataConverter {
   constructor(private preConverter: DataPreConverter) {}
 
-  /** Parse document data (e.g. from a set() call). */
-  parseSetData(
-    methodName: string,
-    input: AnyJs,
-    options: firestore.SetOptions
-  ): ParsedSetData {
+  /** Parse document data from a non-merge set() call.*/
+  parseSetData(methodName: string, input: AnyJs): ParsedSetData {
     const context = new ParseContext(
       UserDataSource.Set,
       methodName,
       FieldPath.EMPTY_PATH
     );
     validatePlainObject('Data must be an object, but it was:', context, input);
 
-    const merge = options.merge !== undefined ? options.merge : false;
+    let updateData = ObjectValue.EMPTY;
+
+    objUtils.forEach(input as Dict<AnyJs>, (key, value) => {
```

Comment:
Why do we need to loop over the top-level keys instead of just calling parseData() directly on the entire object?  Did we clean that up in Android and forget to port?

Reply:
Yeah, I think we forgot to port this back. Fixed.